features/flows/classification/engine/impl/src/test/java/org/opennms/netmgt/flows/classification/internal/value/IpValueTest.java

```diff
@@ -0,0 +1,122 @@
+ * by the Free Software Foundation, either version 3 of the License,
+ * or (at your option) any later version.
+ *
+ * OpenNMS(R) is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with OpenNMS(R).  If not, see:
+ *      http://www.gnu.org/licenses/
+ *
+ * For more information contact:
+ *     OpenNMS(R) Licensing <license@opennms.org>
+ *     http://www.opennms.org/
+ *     http://www.opennms.com/
+ *******************************************************************************/
+
+package org.opennms.netmgt.flows.classification.internal.value;
+
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+import org.opennms.core.network.IPAddress;
+import org.opennms.core.network.IPAddressRange;
+
+public class IpValueTest {
+
+    @Test
+    public void verifyRangedValues() {
+        final IpValue ipValue = new IpValue("10.1.1.1-10.1.1.100");
+        final IPAddressRange range = new IPAddressRange("10.1.1.1", "10.1.1.100");
+        for (IPAddress address : range) {
+            assertThat(ipValue.isInRange(address.toUserString()), is(true));
+        }
+    }
+
+    @Test
+    public void verifySingleValue() {
+        final IpValue ipValue = new IpValue("192.168.0.1");
+        assertThat(ipValue.isInRange("192.168.0.0"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.1"), is(true));
+        assertThat(ipValue.isInRange("192.168.0.2"), is(false));
+    }
+
+    @Test
+    public void verifyMultiValues() {
+        final IpValue ipValue = new IpValue("192.168.0.1, 192.168.0.2, 192.168.0.10");
+        assertThat(ipValue.isInRange("192.168.0.0"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.1"), is(true));
+        assertThat(ipValue.isInRange("192.168.0.2"), is(true));
+        assertThat(ipValue.isInRange("192.168.0.3"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.4"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.5"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.6"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.7"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.8"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.9"), is(false));
+        assertThat(ipValue.isInRange("192.168.0.10"), is(true));
+    }
+
+    @Test
+    public void verifyCIDRValue() {
+        final IpValue ipValue = new IpValue("10.0.0.5,192.168.0.0/24");
+        for (IPAddress ipAddress : new IPAddressRange("192.168.0.0", "192.168.0.255")) {
+            assertThat(ipValue.isInRange(ipAddress.toUserString()), is(true));
+        }
+        assertThat(ipValue.isInRange("10.0.0.5"), is(false));
```

Comment:
Wouldn't this be true? Value is apart of ipValue on declaration.

Reply:
Should also test CIDR values that are not on network address. IE. 10.0.0.0/24 is the same as 10.0.0.1/24 and 10.0.0.255/24

Reply:
Not sure what you mean with the latest comment.

Reply:
Not everyone will use the standard CIDR notation of 192.168.0.0/24, some will use 192.168.0.1/24 instead. Both are identical and valid.

Reply:
Yes, that is true. Internally it is already working as expected, but adding a few more test cases to cover these is a good idea.